module/modmanager/manager.go

```diff
@@ -444,6 +445,19 @@ func (mgr *Manager) newOnUnexpectedExitHandler(mod *module) func(exitCode int) b
 		if mod.inRecovery.Load() {
 			return false
 		}
+
+		// If mod.handles was never set, the module was never actually ready in the
+		// first place before crashing. Log an error and do not attempt a restart;
+		// something is likely wrong with the module implemenation.
+		if mod.handles == nil {
```

Comment:
This is part of a goroutine, so needs locking. You can probably just use the mgr.mu for that rather than adding a new mutex.

Reply:
Ah great catch!